macros/metrics/base/build_in/column_default.sql

```diff
@@ -57,9 +57,38 @@
 {% endmacro %}
 
 {% macro re_data_metric_nulls_percent(column_name) %}
-    {{ re_data_metric_nulls_count(column_name) }} / nullif({{ re_data_metric_row_count() }}, 0) * 100
+    {{ percentage_formula(re_data_metric_nulls_count(column_name), re_data_metric_row_count()) }}
 {% endmacro %}
 
 {% macro re_data_metric_missing_percent(column_name) %}
-    {{ re_data_metric_missing_count(column_name) }} / nullif({{ re_data_metric_row_count() }}, 0) * 100
+    {{ percentage_formula(re_data_metric_missing_count(column_name), re_data_metric_row_count()) }}
+{% endmacro %}
+
+{% macro re_data_metric_regexp_count(column_name, pattern) %}
+    coalesce(
+        sum(
+            case when {{ regexp_match_expression(column_name, pattern) }}
+                then 1
+            else 0
+            end
+        ), 0
+    )
+{% endmacro %}
+
+{% macro re_data_metric_correct_count(column_name, config) %}
+    {% set pattern = config.get('regexp') %}
+    {{ re_data_metric_regexp_count(column_name, pattern) }}
+{% endmacro %}
+
+{% macro re_data_metric_correct_percent(column_name, config) %}
+    {{ percentage_formula(re_data_metric_correct_count(column_name, config), re_data_metric_row_count()) }}
+{% endmacro %}
+
+{% macro re_data_metric_incorrect_count(column_name, config) %}
+    {% set pattern = config.get('regexp') %}
+    {{ re_data_metric_regexp_count(column_name, pattern) }}
+{% endmacro %}
+
+{% macro re_data_metric_incorrect_percent(column_name, config) %}
+    {{ percentage_formula(re_data_metric_incorrect_count(column_name, config), re_data_metric_row_count()) }}
```

Comment:
@mateuszklimek Initially, I created these macros to derive corrent_percent, incorrect_count and, incorrect_percent from the value returned from the `correct_count` but I noticed that it created a dependency on the `correct_count` macro. Not sure if that gives a good user experience.

This current change calculates the value using the regex passed for each metric, it looks more explicit to me and reduces ambiguity.